libbeat/tests/compose/compose.go

```diff
@@ -0,0 +1,238 @@
+package compose
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/libcompose/docker"
+	"github.com/docker/libcompose/docker/ctx"
+	"github.com/docker/libcompose/project"
+	"github.com/docker/libcompose/project/options"
+)
+
+// docker-compose project wrapper
+type composeProject struct {
+	p    project.APIProject
+	file string
+}
+
+type serviceInfo struct {
+	Name    string
+	Running bool
+	Healthy bool
+	// Has been up for more than 1 minute
+	Old bool
+}
+
+// EnsureUp starts all the requested services (must be defined in docker-compose.yml)
+// It returns a tearDown method that must be called once finished, to kill the containers
+func EnsureUp(t *testing.T, services ...string) func() {
+	compose, err := getComposeProject()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, service := range services {
+		err := compose.Start(service)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Kill no longer used containers
+	compose.KillOld(services)
+
+	for _, service := range services {
+		err := compose.Wait(service)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Return down function
+	return func() {}
+}
+
+// Start the container, unless it's running already
+func (c *composeProject) Start(service string) error {
+	c.Lock()
+	defer c.Unlock()
+
+	servicesStatus, err := c.getServices(service)
+	if err != nil {
+		return err
+	}
+
+	if servicesStatus[service] != nil {
+		if servicesStatus[service].Running {
+			// Someone is running it
+			return nil
+		}
+	}
+
+	return c.p.Up(context.Background(), options.Up{
+		Create: options.Create{
+			ForceBuild: true,
+		},
+	}, service)
+}
+
+// Ensure all wanted services are healthy. Wait loop (60s timeout)
+func (c *composeProject) Wait(service string) error {
+	healthy := false
+	seconds := 60
+	for !healthy && seconds > 0 {
+		healthy = true
+
+		servicesStatus, err := c.getServices(service)
+		if err != nil {
+			return err
+		}
+
+		for _, s := range servicesStatus {
+			if !s.Healthy {
+				healthy = false
+				break
+			}
+		}
+
+		time.Sleep(1 * time.Second)
+		seconds--
+	}
+
+	if !healthy {
+		return errors.New("Timeout waiting for container to be healthy")
+	}
+	return nil
+}
+
+func (c *composeProject) Kill(service string) {
+	c.p.Kill(context.Background(), "KILL", service)
+}
+
+func (c *composeProject) KillOld(except []string) error {
+	// Do not kill ourselves :)
+	except = append(except, "beat")
+
+	servicesStatus, err := c.getServices()
+	if err != nil {
+		return err
+	}
+
+	for _, s := range servicesStatus {
+		// Ignore the ones we want
+		if contains(except, s.Name) {
+			continue
+		}
+
+		if s.Old {
+			c.Kill(s.Name)
+		}
+	}
+
+	return nil
+}
+
+// Adquire the lock (30s) timeout
```

Comment:
s/Adquire/Aquire

Probably comment should start with `Lock`to make code style happy.